File: neetcode-150/java/Array-Hashing/twoSum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (m.containsKey(diff))
                return new int[] { m.get(diff), i };
            m.put(nums[i], i);
        }
        return null;
    }
}

File: nexon/01-04-2025/twoSum.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(numbers)):
            diff = target - numbers[i]
            if diff in m:
                return [m[diff], i + 1]
            else:
                m[numbers[i]] = i + 1
        return []
        